src/components/screens/map-screen/map-screen.tsx

```diff
@@ -1660,7 +1692,14 @@ class MapScreen extends React.Component<Props, State> {
    * @param mapViewport a new viewport
    */
   private onViewportChange = async (mapViewport: Viewport) => {
-    this.setState({ mapViewport });
+    var mapBounds = this.mapRef.current?.leafletElement.getBounds();
+    var zoom = this.mapRef.current?.leafletElement.getZoom() as number;
+
+    this.setState({
+      mapBounds,
+      mapViewport,
+      zoom
```

Comment:
```suggestion
    this.setState({
      mapBounds: mapBounds,
      mapViewport: mapViewport,
      zoom: zoom 
```
It should be assigned a value even if names are the same. That is a better readability coding practice